erc/ERCS/erc-7562.md

```diff
@@ -230,63 +329,66 @@ This means that a `Paymaster` and `Factory` contracts cannot practically be an "
+    * Do not add a `UserOperation` to the mempool if the maximum total gas cost, including the new `UserOperation`, is above the deposit of the `paymaster` at the current gas price.
+* **[EREP-015]** A `paymaster` should not have its opsSeen incremented on failure of factory or account
+    * When running 2nd validation (before inclusion in a bundle), if a UserOperation fails because of factory or account error (either a FailOp revert or validation rule), then the paymaster's opsSeen valid is decremented by 1. 
 * **[EREP-020]** A staked factory is "accountable" for account breaking the rules. \
-That is, if the `validateUserOp()` is rejected for any reason, it is treated as if the factory caused this failure, and thus this affects its reputation.
-* **[EREP-030]** A Staked Account is accountable for failures in other entities (`paymaster`, `aggregator`) even if they are staked. \
-(see [https://github.com/eth-infinitism/account-abstraction/issues/274](https://github.com/eth-infinitism/account-abstraction/issues/274))
+  That is, if the `validateUserOp()` is rejected for any reason in a `UserOperation` that has an `initCode`, it is treated as if the factory caused this failure, and thus this affects its reputation.
+* **[EREP-030]** A Staked Account is accountable for failures in other entities (`paymaster`, `aggregator`) even if they are staked.
 * **[EREP-040]** An `aggregator` must be staked, regardless of storage usage.
-* **[EREP-050]** An unstaked `paymaster` may not return a `context`.
-    * TODO: can be removed once we remove the "2nd postOp".
 
 ### Unstaked Entities Reputation Rules
 
 * Definitions:
     * **`opsSeen`, `opsIncluded`, and reputation calculation** are defined above.
-    * `UnstakedReputation` of an entity is a maximum number of entries using this entity allowed in the mempool.
-    * `opsAllowed` is a reputation based calculation for an unstaked entity, representing how many `UserOperations` it is allowed to have in the mempool.
-* **[UREP-010]** `UserOperation` with unstaked sender are only allowed up to `SAME_SENDER_MEMPOOL_COUNT` times in the mempool.
+    * `UnstakedReputation` of an entity determines the maximum number of entries using this entity allowed in the mempool.
+    * `opsAllowed` is a reputation-based calculation for an unstaked entity, representing how many `UserOperations` it is allowed to have in the mempool.
+* **[UREP-010]** An unstaked sender is only allowed to have `SAME_SENDER_MEMPOOL_COUNT` `UserOperation`s  in the mempool. A staked sender is only limited by the SREP rules.
 * **[UREP-020]** For other entities: \
-    `opsAllowed = SAME_UNSTAKED_ENTITY_MEMPOOL_COUNT + (inclusionRate * INCLUSION_RATE_FACTOR) + (min(opsIncluded, 10000)`.
-    * This is a default of `SAME_UNSTAKED_ENTITY_MEMPOOL_COUNT` for new entity
+  `opsAllowed = SAME_UNSTAKED_ENTITY_MEMPOOL_COUNT + (inclusionRate * INCLUSION_RATE_FACTOR) + (min(opsIncluded, 10000)`.
```

Comment:
```suggestion
opsAllowed = SAME_UNSTAKED_ENTITY_MEMPOOL_COUNT + inclusionRate * (INCLUSION_RATE_FACTOR + (min(opsIncluded, 10000))
```

Reply:
What is the role of INCLUSION_RATE_FACTOR here?  It is not really used as a factor, just adds up to 10 to the total number.

- The function starts at 10 (so banning never happens). I guess it's fine if it only protects against a drained paymaster.
- After one inclusion it jumps to 20.  From there it increases by 1 for every inclusion until 10010.
- The drop rate is linear, not faster than the increase rate.  If a paymaster's balance is drained after it had 10000 inclusions, it starts dropping from 10010 to 10 at a slow rate.  It'll drop back to 10 after 10000 unpaid simulations.  Is that the intended behavior?